### ISSUE TITLE
Office.Settings only persist on desktop version but not on web version of Excel 

### ISSUE THREAD

**Issue by ChienHo** [OfficeDev/office-js#5206]
# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web 
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: Version 2411 Build 16.0.18227.20082
* Operating System: Windows 11
* Browser (if using Office on the web): Chrome Version 131.0.6778.140 (Official Build) (64-bit)
 
## Expected behavior

Office.Settings persist when I open the Excel file on the web

## Current behavior

Our Add-in Domo (https://appsource.microsoft.com/en-us/product/office/WA200005264) uses Office.Settings to store some data.
Normally we don't encounter this issue. But sometimes some file has the issue that the web version of Excel doesn't read the data that was stored by the PC version.

## Steps to reproduce

1. Download this file 
[OrderBook_Sales_Domo_IssueFile.xlsx](https://github.com/user-attachments/files/18289115/OrderBook_Sales_Domo_IssueFile.xlsx)
2. Open it on Excel desktop
3. Install our add-in https://appsource.microsoft.com/en-us/product/office/WA200005264
4. Open the add-in
5. Open the dev tool and [add](https://learn.microsoft.com/en-us/javascript/api/office/office.settings?view=common-js-preview#office-office-settings-set-member(1)) some simple data. For example Office.context.document.settings.set('mySetting', {a:1,b:2});
6. [Save](Office.context.document.settings.set('mySetting', 'mySetting value');) the settings
7. Go to Excel on the web and open the same file
8. Open the add-in and devtool
9. Run Office.context.document.settings.get('mySetting')
10. See that nothing is returned

## Context

Our add-in relies on Office.Settings to store data on the file. Without the ability to store data the add-in is basically useless. While this issue doesn't always happen but some of our customers still encounter it and they demand a fix and explanation.

## Useful logs

- [ ] Test file (if only happens on a particular file)
[OrderBook_Sales_Domo_WorkingGood.xlsx](https://github.com/user-attachments/files/18176908/OrderBook_Sales_Domo_WorkingGood.xlsx)

**Comment by RuizhiSunMS** [comment]
Hi @ChienHo, thx for reaching out here. Mark this as #9636028 into our backlog. I will involve our experts to investigate. Will reply here if any update.

**Comment by RuizhiSunMS** [comment]
Hi @ChienHo, would you please help to confirm such things?
1. does your addin use shared runtime?
2. below code is to get office.settings and change it. No matter you change it on web(or client), you can't get the change on the other platform, is it?
```
name: Blank snippet
description: Create a new snippet from a blank template.
host: EXCEL
api_set: {}
script:
  content: |
    $("#run").on("click", () => tryCatch(run));
    $("#run1").on("click", () => tryCatch(run1));

    async function run() {
      await Excel.run(async (context) => {
        console.log(Office.context.document.settings.get("mySetting"));
      });
    }

    async function run1() {
      await Excel.run(async (context) => {
        Office.context.document.settings.set('mySetting', { a: 1, b: 2 });

        await context.sync();
        console.log(12);
      });
    }

    /** Default helper for invoking an action and handling errors. */
    async function tryCatch(callback) {
      try {
        await callback();
      } catch (error) {
        // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
        console.error(error);
      }
    }
  language: typescript
template:
  content: |-
    <button id="run" class="ms-Button">
        <span class="ms-Button-label">Run</span>
    </button>
    <button id="run1" class="ms-Button">
        <span class="ms-Button-label">Run1</span>
    </button>
  language: html
style:
  content: |-
    section.samples {
        margin-top: 20px;
    }

    section.samples .ms-Button, section.setup .ms-Button {
        display: block;
        margin-bottom: 5px;
        margin-left: 20px;
        min-width: 80px;
    }
  language: css
libraries: |
  https://appsforoffice.microsoft.com/lib/1/hosted/office.js
  @types/office-js

  office-ui-fabric-core@11.1.0/dist/css/fabric.min.css
  office-ui-fabric-js@1.5.0/dist/css/fabric.components.min.css

  core-js@2.4.1/client/core.min.js
  @types/core-js

  jquery@3.1.1
  @types/jquery@3.3.1
```
3. how would it be if add above 2 functionalities into your addin?

I'm not very familiar with this part but I suppose it could be related with runtime. I'm reaching out to team owning this to confirm. Please be patient.

**Comment by ChienHo** [comment]
Hi @RuizhiSunMS 

I uploaded the wrong file. This is the file that we've been having issue with:
[OrderBook_Sales_Domo_IssueFile.xlsx](https://github.com/user-attachments/files/18289111/OrderBook_Sales_Domo_IssueFile.xlsx)

To answer your questions:
1. No we don't use shared runtime.

2 & 3. Your code runs fine on scriptlab or localhost (set the data on client and get it on web).
It only has issue with our add-in on the [appstore](https://appsource.microsoft.com/en-us/product/office/WA200005264).

We can't release new build to appstore to test right now but you can still reproduce the issue on the dev tool like what I described above.

**Comment by ChienHo** [comment]
Hi @RuizhiSunMS @guoms1. Is there any updates for this issue? We really rely on this feature.

**Comment by ChienHo** [comment]
Hi @RuizhiSunMS @guoms1 Is there any updates for this issue?

**Comment by RuizhiSunMS** [comment]
@ChienHo, Sorry that here is no update yet

**Comment by amanism19** [comment]
Hi @ChienHo ,
I looked into the issue but wasn’t able to reproduce it in a new file using the add-in you provided. Could you please confirm whether you’re still seeing this issue with a newly created file?

**Comment by ChienHo** [comment]
@amanism19 , yep this issue have fixed on web version, we can close this issue

**Comment by amanism19** [comment]
Based on @ChienHo’s comment, we’re closing this issue.
